Core/Src/main_task.c

```diff
@@ -690,15 +821,15 @@ static void run_downshift_sm(void)
 			if (HAL_GetTick() - begin_exit_gear_tick > DOWNSHIFT_EXIT_TIMEOUT_MS)
 			{
 				// We could not release the gear for some reason. Keep trying anyway
-				downshift_state = ST_D_ENTER_GEAR;
+				next_downshift_state = ST_D_ENTER_GEAR;
 				begin_enter_gear_tick = HAL_GetTick();
 
 				// TODO: Use this in clutchless downshift?
```

Comment:
I would say yes, use thus in clutchless downshift. This is only reached when we fail to move the gear position out of gear and it needs help. This should never be reached in time-based shifting